lib/matplotlib/legend.py

```diff
@@ -1095,26 +1080,12 @@ def _get_anchored_bbox(self, loc, bbox, parentbbox, renderer):
         - parentbbox: a parent box which will contain the bbox. In
             display coordinates.
         """
-        assert loc in range(1, 11)  # called only internally
-
-        BEST, UR, UL, LL, LR, R, CL, CR, LC, UC, C = range(11)
-
-        anchor_coefs = {UR: "NE",
-                        UL: "NW",
-                        LL: "SW",
-                        LR: "SE",
-                        R: "E",
-                        CL: "W",
-                        CR: "E",
-                        LC: "S",
-                        UC: "N",
-                        C: "C"}
 
-        c = anchor_coefs[loc]
+        assert loc in _valid_compass  # as this is called only internally
```

Comment:
Docstring of this internal method needs update on the changed value type of `loc`.